hazelcast-client/src/test/java/com/hazelcast/client/cardinality/ClientCardinalityEstimatorTest.java

```diff
@@ -0,0 +1,220 @@
+
+    @Before
+    public void setup() {
+        hazelcastFactory.newHazelcastInstance();
+        client = hazelcastFactory.newHazelcastClient();
+        estimator = client.getCardinalityEstimator(randomString());
+    }
+
+    @After
+    public void tearDown() {
+        hazelcastFactory.terminateAll();
+    }
+
+    private HazelcastInstance createSerializationConfiguredClient() {
+        final SerializerConfig serializerConfig = new SerializerConfig();
+        serializerConfig.setImplementation(new CustomObjectSerializer());
+        serializerConfig.setTypeClass(CustomObject.class);
+
+        ClientConfig config = new ClientConfig();
+        config.getSerializationConfig().addSerializerConfig(serializerConfig);
+
+        return hazelcastFactory.newHazelcastClient(config);
+    }
+
+    @Test
+    public void test() throws Exception {
+        assertEquals(0, estimator.estimate());
+        assertEquals(true, estimator.aggregate(1L));
+        assertEquals(true, estimator.aggregate(1L));
+        assertEquals(1, estimator.estimate());
+
+        for (long l : new long[] { 2L, 3L, 4L }) {
+            assertEquals(true, estimator.aggregate(l));
+        }
+
+        assertEquals(4, estimator.estimate());
+        assertEquals(true, estimator.aggregate("Test"));
+        assertEquals(5, estimator.estimate());
+    }
+
+    @Test
+    public void testAsync() throws Exception {
+        ICompletableFuture<Long> f1 = estimator.estimateAsync();
+        assertEquals(0L, f1.get().longValue());
+
+        ICompletableFuture<Boolean> f2 = estimator.aggregateAsync(1L);
+        assertEquals(true, f2.get());
+
+        estimator.aggregateAsync(1L).get();
+        f1 = estimator.estimateAsync();
+        assertEquals(1L, f1.get().longValue());
+
+        estimator.aggregateAsync(2L).get();
+        estimator.aggregateAsync(3L).get();
+        estimator.aggregateAsync(4L).get();
+
+        f1 = estimator.estimateAsync();
+        assertEquals(4, f1.get().longValue());
+
+        f2 = estimator.aggregateAsync("Test");
+        assertEquals(true, f2.get());
+
+        estimator.aggregateAsync(1L).get();
+        f1 = estimator.estimateAsync();
+        assertEquals(5L, f1.get().longValue());
+    }
+
+    @Test
+    public void estimate() {
+        CardinalityEstimator estimator = client.getCardinalityEstimator("estimate");
+        assertEquals(0, estimator.estimate());
+    }
+
+    @Test
+    public void estimateAsync()
+            throws ExecutionException, InterruptedException {
+        CardinalityEstimator estimator = client.getCardinalityEstimator("estimateAsync");
+        assertEquals(0, estimator.estimateAsync().get().longValue());
+    }
+
+    @Test
+    public void aggregate() {
+        CardinalityEstimator estimator = client.getCardinalityEstimator("aggregate");
+        assertEquals(true, estimator.aggregate(1L));
+        assertEquals(1L, estimator.estimate());
+    }
+
+    @Test
+    public void aggregateAsync()
```

Comment:
I just throw Exception to get rid of all the exception noise. But that is just flavor.